File: 백준/Bronze/5585. 거스름돈/거스름돈.py
money = [500, 100, 50, 10, 5, 1];

left = 1000 - int(input());

answer = 0;

for i in range(6):
    if left % money[i] >= 0:
        answer += int(left/money[i]);
        left = left % money[i];

print(answer);
    

File: 백준/Silver/14916. 거스름돈/거스름돈.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? 'dev/stdin' : '../input.txt';
const input = fs.readFileSync(filePath).toString().trim().split('\n');

let money = parseInt(input[0]);

let fiveCount = parseInt(money / 5);
let answer = -1;

//만약에 5원으로 나눈 몪이
while (fiveCount != -1) {
  let rest = money - fiveCount * 5;
  if (rest % 2 === 0) {
    answer = fiveCount + parseInt(rest / 2);
    break;
  }
  fiveCount--;
}

console.log(answer);
